mm2src/coins/lp_coins.rs

```diff
@@ -4187,6 +4201,86 @@ pub async fn lp_coinfind_any(ctx: &MmArc, ticker: &str) -> Result<Option<MmCoinS
     Ok(coins.get(ticker).cloned())
 }
 
+pub async fn find_unique_account_ids_any(ctx: &MmArc) -> Result<HashSet<String>, String> {
+    find_unique_account_ids(ctx, false).await
+}
+
+pub async fn find_unique_account_ids_active(ctx: &MmArc) -> Result<HashSet<String>, String> {
+    find_unique_account_ids(ctx, true).await
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn find_unique_account_ids(ctx: &MmArc, _active_only: bool) -> Result<HashSet<String>, String> {
+    // Using a HashSet to ensure uniqueness efficiently
+    let mut account_ids = HashSet::new();
+    // Add default wallet pubkey
+    account_ids.insert(ctx.rmd160_hex());
+
+    Ok(account_ids)
+}
+
+#[cfg(target_arch = "wasm32")]
+async fn find_unique_account_ids(ctx: &MmArc, active_only: bool) -> Result<HashSet<String>, String> {
+    // Using a HashSet to ensure uniqueness efficiently
+    let mut account_ids = HashSet::new();
+    // Add default wallet pubkey
+    account_ids.insert(ctx.rmd160_hex());
+
+    let cctx = try_s!(CoinsContext::from_ctx(ctx));
+    let coins = cctx.coins.lock().await;
+    let coins = coins.values().collect::<Vec<_>>();
+
+    for coin in coins.iter() {
+        if let Some(account) = try_s!(coin.inner.account_db_id()) {
+            if active_only && coin.is_available() {
+                account_ids.insert(account.clone());
+                continue;
+            };
+
+            if !active_only {
+                account_ids.insert(account);
+                continue;
+            }
+        }
+    }
```

Comment:
Knowing that for native and wasm we share the same problem, can we handle both of them with single approach (without target `cfg`s )?

Reply:
yeah sure